src/Arcus.Observability.Tests.Unit/Extensions/TestLoggerExtensions.cs

```diff
@@ -0,0 +1,116 @@
+            IDictionary<string, object> context = GetTelemetryContext(match);
+
+            return new MetricLogEntry(metricName, metricValue, timestamp, context);
+        }
+
+        private static string GetMetricName(Match match)
+        {
+            Assert.True(
+                match.Groups.TryGetValue("metricname", out Group metricNameGroup),
+                "Can't find metric name in logged message");
+            Assert.NotNull(metricNameGroup);
+            
+            string metricName = Assert.Single(metricNameGroup.Captures).Value;
+            return metricName;
+        }
+
+        private static double GetMetricValue(Match match)
+        {
+            Assert.True(
+                match.Groups.TryGetValue("metricvalue", out Group metricValueGroup), 
+                "Can't find metric value in logged message");
+            
+            Assert.NotNull(metricValueGroup);
+            string metricValueText = Assert.Single(metricValueGroup.Captures).Value;
+            
+            Assert.True(
+                double.TryParse(metricValueText, NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture, out double metricValue),
+                "Can't parse metric value to double");
+            
+            return metricValue;
+        }
+
+        private static DateTimeOffset GetTimestamp(Match match)
+        {
+            Assert.True(
+                match.Groups.TryGetValue("timestamp", out Group timestampGroup), 
+                "Can't timestamp in logged message");
+            
+            Assert.NotNull(timestampGroup);
+            string timestampText = Assert.Single(timestampGroup.Captures).Value;
+            
+            Assert.True(
+                DateTimeOffset.TryParse(timestampText, out DateTimeOffset timestamp),
+                "Can't parse timestamp to date time offset");
+           
+            return timestamp;
+        }
+
+        private static IDictionary<string, object> GetTelemetryContext(Match match)
+        {
+            Assert.True(match.Groups.TryGetValue("context", out Group contextGroup), "Can't find context in logged message");
+            Assert.NotNull(contextGroup);
+            string contextText = Assert.Single(contextGroup.Captures).Value;
+
+            char[] trailingChars = {'[', ']', ' '};
+            IDictionary<string, object> context =
+                contextText.Split(';')
+                           .Select(item => item.Split(','))
+                           .ToDictionary(
+                               item => item[0].Trim(trailingChars),
+                               item => (object) item[1].Trim(trailingChars));
+
+            Assert.Single(context, item =>
+            {
+                return item.Key == ContextProperties.General.TelemetryType
+                       && item.Value.ToString() == TelemetryType.Metrics.ToString();
+            });
+            context.Remove(ContextProperties.General.TelemetryType);
```

Comment:
The log entries already add this telemetry type by-default. If we don't remove it, it will be there twice. The assertion above this line will make sure that the telemetry type is there so we don't have to check this in the test.